### PR TITLE
Fix send sync for DnsHandle and Resolver, etc

### PR DESCRIPTION
**Pull request by bluejekyll**
fixes: #456 

edit, issues for the below will be filed so that this can be merged into master ASAP:

- need to clean up the Client library
- need to refactor DnsHandle send return to IntoFuture, to clean up interefaces
- probably other things as well...
- need an example with global Resolver.

### CODE REVIEW

**Comment by zonyitoo** [comment]
Works great in my project. Test code:

```rust
use tokio::executor::current_thread;

lazy_static! {
    static ref GLOBAL_DNS_RESOLVER: ResolverFuture = {
        #[cfg(any(unix, windows))]
        {
            let fut = ResolverFuture::from_system_conf().expect("Failed to create ResolverFuture");
            current_thread::block_on_all(fut).expect("Failed to create DNS resolver")
        }

        #[cfg(not(any(unix, windows)))]
        {
            use trust_dns_resolver::config::{ResolverConfig, ResolverOpts};
            let fut = ResolverFuture::new(ResolverConfig::google(), ResolverOpts::default());
            current_thread::block_on_all(fut).expect("Failed to create DNS resolver")
        }
    };
}

pub fn resolve(config: Arc<Config>, addr: &str, port: u16, check_forbidden: bool) -> IoFuture<Vec<SocketAddr>> {
    let owned_addr = addr.to_owned();
    let owned_addr2 = owned_addr.clone();
    let fut = GLOBAL_DNS_RESOLVER.lookup_ip(addr)
                                 .map_err(move |err| {
                                              error!("Failed to resolve {}, err: {}", owned_addr2, err);
                                              io::Error::new(io::ErrorKind::Other, "dns resolve error")
                                          })
                                 .and_then(move |lookup_result| {
                                     // ...
                                 });
```

**Comment by bluejekyll** [comment]
Thanks for the example!

I still have to get this compiling in the entire project, and then some cleanup, but I’m really happy this works for you!

File: resolver/examples/global_resolver.rs [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+            }
+
+            // For other operating systems, we can use one of the preconfigured definitions
+            #[cfg(not(any(unix, windows)))]
+            {
+                // Directly reference the config types
+                use trust_dns_resolver::config::{ResolverConfig, ResolverOpts};
+
+                // Get a new resolver with the google nameservers as the upstream recursive resolvers
+                future = ResolverFuture::new(ResolverConfig::google(), ResolverOpts::default()).expect("Failed to create ResolverFuture");
```
Comment (zonyitoo) [inline resolver/examples/global_resolver.rs]: BUG. `ResolverFuture::new` doesn't return a `Result`

Comment (bluejekyll) [reply resolver/examples/global_resolver.rs]: Gah. I forgot to add this to the appveyor build. Would have caught this.

Thanks.

**Pull request merged**